File: .sdk-repos/version-8.9/orchestration-cluster-api-csharp/examples/Document.cs
// Compilable usage examples for document operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class DocumentExamples
{
    #region CreateDocumentLink
    // <CreateDocumentLink>
    public static async Task CreateDocumentLinkExample(DocumentId documentId)
    {
        using var client = CamundaClient.Create();

        var result = await client.CreateDocumentLinkAsync(
            documentId,
            new DocumentLinkRequest());

        Console.WriteLine($"Document link: {result.Url}");
    }
    // </CreateDocumentLink>
    #endregion CreateDocumentLink

    #region DeleteDocument

    // <DeleteDocument>
    public static async Task DeleteDocumentExample(DocumentId documentId)
    {
        using var client = CamundaClient.Create();

        await client.DeleteDocumentAsync(documentId);
    }
    // </DeleteDocument>
    #endregion DeleteDocument

    #region CreateDocument

    // <CreateDocument>
    public static async Task CreateDocumentExample()
    {
        using var client = CamundaClient.Create();

        using var content = new MultipartFormDataContent();
        content.Add(new ByteArrayContent(System.Text.Encoding.UTF8.GetBytes("Hello, world!")), "file", "hello.txt");

        var result = await client.CreateDocumentAsync(content);

        Console.WriteLine($"Document ID: {result.DocumentId}");
    }
    // </CreateDocument>
    #endregion CreateDocument

    #region CreateDocuments

    // <CreateDocuments>
    public static async Task CreateDocumentsExample()
    {
        using var client = CamundaClient.Create();

        using var content = new MultipartFormDataContent();
        content.Add(new ByteArrayContent(System.Text.Encoding.UTF8.GetBytes("File one")), "files", "one.txt");
        content.Add(new ByteArrayContent(System.Text.Encoding.UTF8.GetBytes("File two")), "files", "two.txt");

        var result = await client.CreateDocumentsAsync(content);

        foreach (var doc in result.CreatedDocuments)
        {
            Console.WriteLine($"Created: {doc.DocumentId}");
        }
    }
    // </CreateDocuments>
    #endregion CreateDocuments

    #region GetDocument

    // <GetDocument>
    public static async Task GetDocumentExample(DocumentId documentId)
    {
        using var client = CamundaClient.Create();

        var content = await client.GetDocumentAsync(documentId);

        Console.WriteLine($"Downloaded document: {documentId}");
    }
    // </GetDocument>
    #endregion GetDocument
}

File: .sdk-repos/orchestration-cluster-api-go/examples/document.go
// Document operations: upload, download, link, and delete documents.
package examples

import (
	"context"
	"fmt"

	camunda "github.com/camunda/orchestration-cluster-api-go"
	openapi "github.com/camunda/orchestration-cluster-api-go/client"
)

func createDocumentExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region CreateDocument
	// The document payload is attached via request options (functional opts) or
	// the Raw() client; here we call the ergonomic facade method.
	ref, err := client.CreateDocument(ctx)
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", ref)
	// endregion CreateDocument
	return nil
}

func createDocumentsExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region CreateDocuments
	// Batch upload multiple documents in a single multipart request.
	result, err := client.CreateDocuments(ctx)
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", result)
	// endregion CreateDocuments
	return nil
}

func getDocumentExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region GetDocument
	file, err := client.GetDocument(ctx, "doc-123")
	if err != nil {
		return err
	}
	fmt.Printf("downloaded to %s\n", file.Name())
	// endregion GetDocument
	return nil
}

func deleteDocumentExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region DeleteDocument
	return client.DeleteDocument(ctx, "doc-123")
	// endregion DeleteDocument
}

func createDocumentLinkExample(ctx context.Context, client *camunda.CamundaClient) error {
	// region CreateDocumentLink
	// Create a short-lived, shareable download link for a stored document.
	link, err := client.CreateDocumentLink(ctx, "doc-123", *openapi.NewDocumentLinkRequest())
	if err != nil {
		return err
	}
	fmt.Printf("%v\n", link)
	// endregion CreateDocumentLink
	return nil
}
